__tests__/actions.test.ts

```diff
@@ -74,6 +81,48 @@ it("runs command actions", async () => {
   expect(typeof mockedExec.mock.calls[0][1].env.GM_DBURL).toBe("string");
 });
 
+it("runs sql afterReset action with correct connection string when root", async () => {
+  mockFs({
+    "migrations/sqlfile1.sql": `[CONTENT:migrations/sqlfile1.sql]`,
+  });
+  const parsedSettings = await parseSettings({
+    connectionString: TEST_DATABASE_URL,
+    afterReset: [{ _: "sql", file: "sqlfile1.sql", root: true }],
+  });
+  const mockedWithClient: jest.Mock<typeof withClient> = withClient as any;
+  mockedWithClient.mockClear();
+  await executeActions(parsedSettings, false, parsedSettings.afterReset);
+  expect(mockedWithClient).toHaveBeenCalledTimes(1);
+  expect(mockedWithClient.mock.calls[0][0]).toBe(TEST_DATABASE_NAME);
+});
+
+it("runs command afterReset action with correct env vars when root", async () => {
+  const parsedSettings = await parseSettings({
+    connectionString: TEST_DATABASE_URL,
+    rootConnectionString: TEST_ROOT_DATABASE_URL,
+    afterReset: [
+      { _: "command", command: "touch testCommandAction", root: true },
+    ],
+  });
+  const mockedExec: jest.Mock<typeof exec> = exec as any;
+  mockedExec.mockClear();
+  mockedExec.mockImplementationOnce((_cmd, _options, callback) =>
+    callback(null, { stdout: "", stderr: "" }),
+  );
+
+  await executeActions(parsedSettings, false, parsedSettings.afterReset);
+  // When `root: true`, GM_DBUSER may be perceived as ambiguous, so we must not set it.
+  expect(mockedExec.mock.calls[0][1].env.GM_DBUSER).toBe(undefined);
+  const connectionStringParts = parse(TEST_DATABASE_URL);
+  const rootConnectionStringParts = parse(TEST_ROOT_DATABASE_URL);
```

Comment:
Just to make the test clearer, we should assert that the databases were previously different.

```suggestion
  const rootConnectionStringParts = parse(TEST_ROOT_DATABASE_URL);
  expect(rootConnectionStringParts.database).not.toBe(connectionStringParts.database);
```